fpgalib/test/test_dac.py

```diff
@@ -0,0 +1,95 @@
+"""This is intended to test fpgalib/dac.py"""
+
+import pytest
+import fpgalib.dac as dac
+
+
+class TestDAC15(object):
+    @classmethod
+    def setup_class(cls):
+        cls.dac = dac.DAC_Build15(0, 'dac_name')
+
+        # sequence SRAM: pi pulse from 0-32, pi pulse from 32-64, ro from 64-400
+        # experiment: first pi -> measure -> second pi -> measure -> end
+        cls.first_pi_start = 0
+        cls.first_pi_end = 32
+        cls.second_pi_start = 32
+        cls.second_pi_end = 64
+        cls.ro_start = 64
+        cls.ro_end = 400
+
+        cls.jump_args = [
+            ('JUMP', [cls.first_pi_end, cls.ro_start, 1]),
+            ('JUMP', [cls.ro_end, cls.second_pi_start, 2]),
+            ('END', [cls.ro_end])
+        ]
+
+    def make_entries(self, jump_args):
+        return [self.dac.make_jump_table_entry(name, args)
+                for name, args in jump_args]
+
+    def test_make_jump_table_entry(self):
+        _ = self.make_entries(self.jump_args)
+        # we made it this far and we're feeling pretty good about it
+
+    def test_make_jump_table(self):
+        entries = self.make_entries(self.jump_args)
+        self.dac.make_jump_table(entries)
+        # we made it this far and we're feeling pretty good about it
+
+    def test_catch_bad_jump_table(self):
+        """Test jump table commands that are executed too frequently"""
+
+        # from_addr of 16 followed by 20 are too close together
+        bad_jump_args = [
+            ('NOP', [16]),
+            ('NOP', [20]),
+            ('END', [400])
+        ]
```

Comment:
I think it's best here to split these cases into separate test methods and give them descriptive names rather than having multiple cases within one methods with comments to explain what's different. I also generally think that if test methods are small, test one thing, and named descriptively, then docstrings are usually not needed on each method.
```python
def test_nop_nop_from_addr_too_close(self):
    entries = self.make_entries([
        ('NOP', [16]),
        ('NOP', [20]),
        ('END', [400])
    ])
    with pytest.raises(AssertionError):
        self.dac.make_jump_table(entries)

def test_jump_nop_from_addr_too_close(self):
    ...
```
Obviously this slightly increases the amount of boilerplate, but I think it makes debugging test failures much easier.